### PR TITLE
cleanup to support provider's custom ssh command runner

### PR DESCRIPTION
**Pull request by AmeerHajAli**
## Why are these changes needed?
This allows the autoscaler provider to support custom ssh command runners.

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/autoscaler/kubernetes/node_provider.py [hunk 1]
```diff
@@ -85,3 +86,9 @@ def terminate_node(self, node_id):
     def terminate_nodes(self, node_ids):
         for node_id in node_ids:
             self.terminate_node(node_id)
+
+    def get_command_runner(self, log_prefix, node_id, provider, auth_config,
+                           cluster_name, process_runner, use_internal_ip):
+        return KubernetesCommandRunner(log_prefix, node_id, provider,
```
Comment (ericl) [inline python/ray/autoscaler/kubernetes/node_provider.py:92]: provider == self right? So we could omit this.

File: python/ray/autoscaler/node_provider.py [hunk 2]
```diff
@@ -208,3 +210,9 @@ def terminate_nodes(self, node_ids):
     def cleanup(self):
         """Clean-up when a Provider is no longer required."""
         pass
+
+    def get_command_runner(self, log_prefix, node_id, provider, auth_config,
+                           cluster_name, process_runner, use_internal_ip):
+        """ Returns the CommandRunner class used to perform SSH commands. """
```
Comment (ericl) [inline python/ray/autoscaler/node_provider.py:216]: Can you document the arguments in pydoc format:
"""

Args:
  log_prefix (str): What this is...
  ...
"""
?

**Review by ericl — CHANGES_REQUESTED** [review]

File: python/ray/autoscaler/node_provider.py (hunk 2 above)
Comment (AmeerHajAli) [reply python/ray/autoscaler/node_provider.py:216]: Shouldn't this be documented in the class definitions?

File: python/ray/autoscaler/kubernetes/node_provider.py (hunk 1 above)
Comment (AmeerHajAli) [reply python/ray/autoscaler/kubernetes/node_provider.py:92]: yes

**Review by ericl — APPROVED** [review]

File: python/ray/autoscaler/updater.py [hunk 3]
```diff
@@ -35,13 +35,15 @@ def with_interactive(cmd):
 
 
 class KubernetesCommandRunner:
-    def __init__(self, log_prefix, namespace, node_id, auth_config,
-                 process_runner):
+    def __init__(self, log_prefix, node_id, provider, auth_config,
```
Comment (edoakes) [inline python/ray/autoscaler/updater.py:38]: Any reason why we're passing in the full node provider here just to access the namespace? Can we just pass in the namespace instead?

**Comment by ericl** [comment]
Lint job is failing in travis.

File: python/ray/autoscaler/updater.py (hunk 3 above)
Comment (edoakes) [reply python/ray/autoscaler/updater.py:38]: Sure, but why do we need to change the signature for this class constructor?

Comment (ericl) [reply python/ray/autoscaler/updater.py:38]: That's a good point, the constructor signature of the actual cmd runner doesn't matter, only the get_command_runner() interface.

Comment (AmeerHajAli) [reply python/ray/autoscaler/updater.py]: ok I fixed it.

**Pull request merged**